src/layers/S2-Grid-Layer.ts

```diff
@@ -10,6 +10,7 @@ import VectorLayer from 'ol/layer/Vector.js'
 import { transformExtent } from 'ol/proj'
 import VectorSource from 'ol/source/Vector.js'
 import { getArea } from 'ol/sphere'
+import { getArea as getAreaExtent } from 'ol/extent'
```

Comment:
It looks like you're not using this import, and it's good that you aren't: the result would be the area calculated for cartesian coordinates, which would be very wrong for Web Mercator or geographic coordinates. The correct way is to use `getArea` from `ol/sphere`, like it is currently done.